src/Stratis.Bitcoin.Tests/Utilities/AsyncManualResetEventTest.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.Utilities
+{
+    public class AsyncManualResetEventTest
+    {
+        [Fact]
+        public async void AsyncManualResetEvent_WaitAsync()
+        {
+            AsyncManualResetEvent manualResetEvent = new AsyncManualResetEvent(false);
+
+            Task task = Task.Run(async delegate
+            {
+                await Task.Delay(500);
+                manualResetEvent.Set();
+            });
+
+            await manualResetEvent.WaitAsync();
+        }
+
+        [Fact]
+        public void AsyncManualResetEvent_CanReset()
```

Comment:
we could have CanSetAndReset and test setting the event again and checking ...

Reply:
Fixed.